### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by aliotta**

### CODE REVIEW

File: analytics_plugin.py [hunk 1]
```diff
@@ -42,32 +41,36 @@ def tasks_report(session) -> Any:
         raise e
     else:
         # remove the dummy operator and the astronomer_monitoring_dag that is added in runtime from task count
-        query = (
```
Comment (aliotta) [inline analytics_plugin.py:45]: moved this to its own method such that I can mock out it and only it.

**Comment by rishkarajgi** [comment]
Could you please update the README.md with instructions on how to run the tests?

File: analytics_plugin.py [hunk 2]
```diff
@@ -30,8 +30,7 @@
 airflow_webserver_base_url = configuration.get("webserver", "BASE_URL")
 
 
-@provide_session
-def tasks_report(session) -> Any:
+def tasks_report() -> Any:
     untrusted_start_date = request.args.get("startDate")  # 2022-08-01
     untrusted_end_date = request.args.get("endDate")  # 2022-08-30
     try:
```
Comment (blag) [inline analytics_plugin.py:36]: Does it make sense to have a separate check, one for the start date and one for the end date? That way the logged error includes that detail: which parameter didn't validate. Seems like cleaner code to me.

**Comment by bote795** [comment]
tested locally and does work

**Pull request merged**